python/cudf/cudf/_lib/column.pxd

```diff
@@ -22,10 +25,12 @@ cdef class Column:
         cdef object _data
         cdef object _mask
         cdef object _null_count
+        cdef object _distinct_count
 
     cdef column_view _view(self, size_type null_count) except *
     cdef column_view view(self) except *
     cdef mutable_column_view mutable_view(self) except *
+    cpdef pylibcudf.Column to_pylibcudf(self, mode: Literal["read", "write"])
```

Comment:
What concepts in libcudf do `"read"` and `"write"` translate to? Also, is a boolean more appropriate than a literal?

Reply:
I don't like that we need this parameter. It has nothing to do with libcudf, and everything to do with cudf Python. It maps directly to the `Buffer.get_ptr` function and is necessary so that we access the pointers in the correct way to manage exposure. I don't see a good way around this, but even if there is a better approach I think it makes sense to wait on implementing more complexity until the COW/spilling unification is completed.

To be clear, I don't foresee exposing this functionality at all in a public `to_pylibcudf` function. If a user calls e.g. `Series.to_pylibcudf` we should always assume that it's exposed. This functionality is purely internal to simplify the mapping of the existing cudf Cython internal logic.